tools/dds/dds-server/dds-server.cpp

```diff
@@ -17,11 +17,39 @@ using namespace tools;
 
 dds_server::dds_server()
     : _running( false )
+    , _trigger_msg_send ( false )
     , _participant( nullptr )
     , _publisher( nullptr )
     , _topic( nullptr )
     , _topic_type( new librealsense::dds::topics::device_info::type )
     , _dds_device_dispatcher( 10 )
+    , _new_client_handler( [this]( dispatcher::cancellable_timer timer ) {
+
+        // We wait until the new reader callback indicate a new reader has joined or until the
+        // active object is stopped
+        std::unique_lock< std::mutex > lock( _new_client_mutex );
+        _new_client_cv.wait( lock, [this, timer]() {
+            return _trigger_msg_send.load() || timer.was_stopped();
+        } );
+        if( _new_client_handler.is_active() && _trigger_msg_send.load() )
+        {
+            _trigger_msg_send = false;
+            _dds_device_dispatcher.invoke( [this]( dispatcher::cancellable_timer ) {
+                for( auto dev_writer : _device_handle_by_sn )
```

Comment:
Is the access to `_device_handle_by_sn` protected somehow? What I'm concerned about is that someone will try modifying it while we're trying to go over it...

Reply:
If I didn't missed something all of the access to _device_handle_by_sn  is inside the dispatcher, meaning it is protected by design.